e2e/testsuite/diagnostics/diagnostics.go

```diff
@@ -0,0 +1,157 @@
+		}
+
+		t.Logf("successfully wrote log file %s", logFile)
+		diagnosticFiles := chainDiagnosticAbsoluteFilePaths(*cfg.ChainAConfig)
+		diagnosticFiles = append(diagnosticFiles, chainDiagnosticAbsoluteFilePaths(*cfg.ChainBConfig)...)
+
+		for _, absoluteFilePathInContainer := range diagnosticFiles {
+			localFilePath := ospath.Join(containerDir, ospath.Base(absoluteFilePathInContainer))
+			if err := fetchAndWriteDiagnosticsFile(ctx, dc, container.ID, localFilePath, absoluteFilePathInContainer); err != nil {
+				t.Logf("failed to fetch and write file %s for container %s in test cleanup: %s", absoluteFilePathInContainer, containerName, err)
+				continue
+			}
+			t.Logf("successfully wrote diagnostics file %s", absoluteFilePathInContainer)
+		}
+	}
+}
+
+// getContainerName returns a either the ID of the container or stripped down human-readable
+// version of the name if the name is non-empty.
+//
+// Note: You should still always use the ID  when interacting with the docker client.
+func getContainerName(t *testing.T, container dockertypes.Container) string {
+	t.Helper()
+	// container will always have an id, by may not have a name.
+	containerName := container.ID
+	if len(container.Names) > 0 {
+		containerName = container.Names[0]
+		// remove the test name from the container as the folder structure will provide this
+		// information already.
+		containerName = strings.TrimRight(containerName, "-"+t.Name())
+		containerName = strings.TrimLeft(containerName, "/")
+	}
+	return containerName
+}
+
+// fetchAndWriteDiagnosticsFile fetches the contents of a single file from the given container id and writes
+// the contents of the file to a local path provided.
+func fetchAndWriteDiagnosticsFile(ctx context.Context, dc *dockerclient.Client, containerID, localPath, absoluteFilePathInContainer string) error {
+	fileBz, err := dockerutil.GetFileContentsFromContainer(ctx, dc, containerID, absoluteFilePathInContainer)
+	if err != nil {
+		return err
+	}
+
+	filePath := ospath.Join(localPath)
+	if err := os.WriteFile(filePath, fileBz, 0750); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// chainDiagnosticAbsoluteFilePaths returns a slice of absolute file paths (in the containers) which are the files that should be
+// copied locally when fetching diagnostics.
+func chainDiagnosticAbsoluteFilePaths(cfg ibc.ChainConfig) []string {
+	return []string{
+		fmt.Sprintf("/var/cosmos-chain/%s/config/genesis.json", cfg.Name),
+		fmt.Sprintf("/var/cosmos-chain/%s/config/app.toml", cfg.Name),
+		fmt.Sprintf("/var/cosmos-chain/%s/config/config.toml", cfg.Name),
+		fmt.Sprintf("/var/cosmos-chain/%s/config/client.toml", cfg.Name),
+	}
+}
+
+// getE2EDir finds the e2e directory above the test.
+func getE2EDir(t *testing.T) (string, error) {
+	t.Helper()
+
+	wd, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+
+	const maxAttempts = 100
+	count := 0
+	for ; !strings.HasSuffix(wd, e2eDir) || count > maxAttempts; wd = ospath.Dir(wd) {
+		count++
+	}
```

Comment:
I'm a bit confused by this. Is it possible to run these e2e's when your wd isn't the e2e directory?

Reply:
this is not strictly necessary, I just wanted to make sure that the test errors out if for some reason this case is ever true instead of hanging forever. It's probably safe to remove this entirely but I just went the extra careful route